adagrams/game.py

```diff
@@ -1,11 +1,135 @@
+import random
+
 def draw_letters():
-    pass
+ 
+    NUM_TILES = 10
+    letters_grabbed = []
+    letters_remaining_pool = LIST_OF_LETTERS.copy()
+    
+    for i in range(NUM_TILES):
+        # update the last index of the remaining pool list that changes everytime
+        # random.randint stop is inclusive, so -1 which reflect the size of the pool
+        index = len(letters_remaining_pool)-1
+        # pop() allows to return the letter that was taken out and update the remaining pool list
+        letter_grabbed = letters_remaining_pool.pop(random.randint(0,index))
+        letters_grabbed.append(letter_grabbed)
+
+    return(letters_grabbed)
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    word = word.upper()
+    letter_bank_remaining = letter_bank.copy()
+
+    for letter in word:
+        # prevent the tiles that are used being used again ie "DOOR" cannot be made with D,O,R,X,X,X
+        if letter in letter_bank_remaining:
+            letter_bank_remaining.remove(letter)
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    
+    score = 0
+    word = word.upper()
+    for letter in word:
+        # score saved as values to the keys(letter)
+        score += VALUES_DICT[letter]
+    if len(word) >= 7:
+        score += 8
```

Comment:
While a hand can't be more than 10 letters and therefore a word can't be more than 10 letters, you could make this check more explicit and conform to the instructions in the README more by having the check on line 40 be: `if 7 <= len(word) <= 10:`

Reply:
Using constant variables will make your code more self-documenting and easier to understand. If I don't know anything about this game, I'm left wondering what the significance of the integer literals are. 

Consider something like:
```python
BONUS_LENGTH_MIN = 7
BONUS_LENGTH_MAX = 10
BONUS_POINTS = 8

if BONUS_LENGTH_MIN <= len(word) <= BONUS_LENGTH_MAX:
        score += BONUS_POINTS
```